apis/python/src/tiledbsoma/util_arrow.py

```diff
@@ -121,36 +121,31 @@ def get_arrow_schema_from_tiledb_uri(
     return pa.schema(arrow_schema_dict)
 
 
-def ascii_to_unicode_pyarrow_readback(record_batch: pa.RecordBatch) -> pa.RecordBatch:
+def ascii_to_unicode_pyarrow_readback(table: pa.Table) -> pa.Table:
     """
     Implements the 'decode on read' part of our ASCII/Unicode logic
     """
     # TODO: COMMENT/LINK HEAVILY
-    names = [ofield.name for ofield in record_batch.schema]
+    names = [ofield.name for ofield in table.schema]
     new_fields = []
     for name in names:
-        old_field = record_batch[name]
-        if isinstance(old_field, pa.LargeBinaryArray):
+        old_field = table[name]
+        if len(old_field) > 0 and isinstance(old_field[0], pa.LargeBinaryScalar):
             nfield = pa.array(
                 [element.as_py().decode("utf-8") for element in old_field]
             )
             new_fields.append(nfield)
         else:
             new_fields.append(old_field)
-    return pa.RecordBatch.from_arrays(new_fields, names=names)
+    return pa.Table.from_arrays(new_fields, names=names)
 
 
-def concat_batches(batch_generator: Iterator[Any]) -> pa.RecordBatch:
+def concat_tables(table_generator: Iterator[Any]) -> pa.Table:
```

Comment:
This routine is obsolete  and can be removed.  You can directly use pyarrow.concat_tables -- it accepts an iterator, so there is no benefit to creating a list to then hand to pyarrow.concat_tables.

In fact, there is a downside - this code requires that all tables be fully materialized prior to concatenation.